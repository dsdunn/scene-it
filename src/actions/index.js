export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
});

export const hasErrored = (bool = false) => ({
  type: 'HAS_ERRORED',
  hasErrored: bool
});

export const eventsFetchSuccess = (events) => ({
  type: 'EVENTS_FETCH_SUCCESS',
  events: events
});

export const locationFetchSuccess = (location) => ({
  type: 'LOCATION_FETCH_SUCCESS',
  location: location
});

export const selectEvent = (event) => ({
  type: "SELECT_EVENT",
  event: event
})

export const unselectEvent = () => ({
  type: "UNSELECT_EVENT"
})

