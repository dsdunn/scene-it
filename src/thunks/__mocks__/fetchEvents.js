export const fetchEvents = jest.fn()