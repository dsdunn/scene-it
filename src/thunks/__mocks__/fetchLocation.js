export const fetchLocation = jest.fn();