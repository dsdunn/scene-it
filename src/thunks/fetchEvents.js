import { isLoading, hasErrored, eventsFetchSuccess } from '../actions';
import { dataCleaner } from '../helper';

export const fetchEvents = (url) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true));
      const response = await fetch(url,{mode:"cors"});
      if (!response.ok) {
        throw Error(response.statusText);
      }
      dispatch(isLoading(false));
      const result = await response.json();
      const cleanData = dataCleaner(result.events.event);
      dispatch(eventsFetchSuccess(cleanData));
      console.log(result.events.event);
    } catch (error) {
      dispatch(hasErrored(true));
    }
  };
};