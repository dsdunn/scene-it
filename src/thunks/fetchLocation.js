import { hasErrored, locationFetchSuccess } from '../actions';
import { mapKey } from '../apiKey';

export const fetchLocation = (location) => {
  const url = `https://maps.googleapis.com/maps/api/geocode/json?address=${location}&key=${mapKey}`
  return async (dispatch) => {
    try{
      const response = await fetch(url);
      if (!response.ok) {
        throw Error(response.statusText);
      }
      const result = await response.json();
      const coordinates = result.results[0].geometry.location;
      dispatch(locationFetchSuccess(coordinates));
    } catch (error) {
      dispatch(hasErrored(true));
    }
  };
};