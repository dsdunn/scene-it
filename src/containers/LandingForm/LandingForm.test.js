import React from 'react';
import { shallow, mount } from 'enzyme';
import { LandingForm } from './index';
import { mapStateToProps, mapDispatchToProps } from './index';
import { fetchEvents } from '../../thunks/fetchEvents';
import { fetchLocation } from '../../thunks/fetchLocation';
import { locationFetchSuccess } from '../../actions';

describe('BodyForm', () => {

  const mockFetchEvents = jest.fn().mockImplementation(() => Promise.resolve({status: 'ok'}));
  const mockFetchLocation = jest.fn().mockImplementation(() => Promise.resolve({status: 'ok'}));
  const mockSetLocation = jest.fn().mockImplementation(() => Promise.resolve({status: 'ok'}));
  const mockEvent = {
    preventDefault: jest.fn(),
    target: {id: 'status', value: 'ok, I guess'}
  }
  const mockProps = {
    events: ['one', 'two', 'three'],
    selectedEvent: {eventId: 1},
    location: {lat:1, lng:2},
    fetchEvents: mockFetchEvents,
    fetchLocation: mockFetchLocation,
    setLocation: mockSetLocation,
    history: []
  }

  const mockGeolocation = {getCurrentPosition: jest.fn()};

  let wrapper;

  beforeEach(()=>{
    wrapper = shallow(<LandingForm {...mockProps} />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should call handleChange on text input change', () => {
    wrapper = mount(<LandingForm {...mockProps} />);
    const spy = jest.spyOn(wrapper.instance(), 'handleChange');

    wrapper.instance().forceUpdate();
    wrapper.find('#locationName').simulate('change', mockEvent);
    expect(spy).toHaveBeenCalled();
  })

  it('should update state when handleChange is called', () => {
    wrapper.instance().handleChange(mockEvent)
    expect(wrapper.state('status')).toEqual('ok, I guess');
  })

  it('should call handleSubmit on form submission', () => {
    wrapper = mount(<LandingForm {...mockProps} />);
    const spy = jest.spyOn(wrapper.instance(), 'handleSubmit');

    wrapper.instance().forceUpdate();
    wrapper.find('form').simulate('submit', mockEvent);
    expect(spy).toBeCalled();
  })

  it('should call mockFetchLocation when handleSubmit is called', async () => {
    await wrapper.instance().handleSubmit(mockEvent);
    expect(mockFetchLocation).toBeCalled();
  })

  it('should call mockFetchEvents when handleSubmit is called', async () => {
    await wrapper.instance().handleSubmit(mockEvent);
    expect(mockFetchEvents).toBeCalled();
  })

  it('should call useCurrent when checkbox is checked', () => {
    wrapper = mount(<LandingForm {...mockProps} />);
    const spy = jest.spyOn(wrapper.instance(), 'useCurrent');

    wrapper.instance().forceUpdate();
    wrapper.find('#use-current-location').simulate('change', mockEvent);
    expect(spy).toBeCalled();
  })

  it('should update the state when useCurrent is called', () => {
    global.navigator.geolocation = mockGeolocation;
    wrapper.instance().useCurrent(mockEvent);
    expect(wrapper.state('useCurrent')).toEqual(true);
  })

})

describe('mapStateToProps', () => {
  it('should return an object with a location', () => {
    const mockState = {
      events: ['one', 'two', 'three'],
      randomKey: 'this shouldn\'t show up',
      selectedEvent: {eventId: 1},
      location: {lat: 1, lng: 2}
    }
    const expected = {
      location: {lat: 1, lng: 2}
    }
    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected);
  })
})


describe('mapDispatchToProps', () => {

  it('calls dispatch with fetchEvents action when fetchEvents is called', () => {
    const mockDispatch = jest.fn();
    const mockUrl = 'someUrl';
    const actionToDispatch = fetchEvents(mockUrl);
    const mappedProps = mapDispatchToProps(mockDispatch);

    mappedProps.fetchEvents(mockUrl);
    expect(mockDispatch).toBeCalledWith(actionToDispatch)
  })

  it('calls dispatch with fetchLocation action when fetchLocation is called', () => {
    const mockDispatch = jest.fn();
    const mockUrl = 'someUrl';
    const actionToDispatch = fetchLocation(mockUrl);
    const mappedProps = mapDispatchToProps(mockDispatch);

    mappedProps.fetchLocation(mockUrl);
    expect(mockDispatch).toBeCalledWith(actionToDispatch)
  })

  it('calls dispatch with locationFetchSuccess action when setLocation is called', () => {
    const mockDispatch = jest.fn();
    const mockLocation = 'someLocation';
    const actionToDispatch = locationFetchSuccess(mockLocation);
    const mappedProps = mapDispatchToProps(mockDispatch);

    mappedProps.setLocation(mockLocation);
    expect(mockDispatch).toBeCalledWith(actionToDispatch)
  })
})




