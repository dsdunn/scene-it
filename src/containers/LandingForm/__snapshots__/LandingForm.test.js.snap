// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyForm should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LandingForm
    events={
        Array [
            "one",
            "two",
            "three",
          ]
    }
    fetchEvents={[Function]}
    fetchLocation={[Function]}
    history={Array []}
    location={
        Object {
            "lat": 1,
            "lng": 2,
          }
    }
    selectedEvent={
        Object {
            "eventId": 1,
          }
    }
    setLocation={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="landing-form"
        onSubmit={[Function]}
>
        <header
                className="App-header"
        >
                <h1
                        className="App-title"
                >
                        Scene-It
                </h1>
                <h5
                        className="header-description"
                >
                        Enter a location or check "use my location" to find out what's happening in your scene this week!
                </h5>
        </header>
        <div
                className="user-select"
        >
                <input
                        id="use-current-location"
                        onChange={[Function]}
                        type="checkbox"
                />
                <label
                        htmlFor="use-current-location"
                >
                        Use my current location
                </label>
        </div>
        <div
                className="query-info"
        >
                <input
                        id="locationName"
                        onChange={[Function]}
                        placeholder="location"
                        value=""
                />
                <input
                        id="keywords"
                        onChange={[Function]}
                        placeholder="keywords (optional)"
                />
                <button
                        className="landing-submit"
                        disabled={true}
                >
                        Get Events
                </button>
        </div>
</form>,
      "className": "landing-body",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <h1
                        className="App-title"
            >
                        Scene-It
            </h1>
            <h5
                        className="header-description"
            >
                        Enter a location or check "use my location" to find out what's happening in your scene this week!
            </h5>
</header>,
          <div
            className="user-select"
>
            <input
                        id="use-current-location"
                        onChange={[Function]}
                        type="checkbox"
            />
            <label
                        htmlFor="use-current-location"
            >
                        Use my current location
            </label>
</div>,
          <div
            className="query-info"
>
            <input
                        id="locationName"
                        onChange={[Function]}
                        placeholder="location"
                        value=""
            />
            <input
                        id="keywords"
                        onChange={[Function]}
                        placeholder="keywords (optional)"
            />
            <button
                        className="landing-submit"
                        disabled={true}
            >
                        Get Events
            </button>
</div>,
        ],
        "className": "landing-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="App-title"
>
                Scene-It
</h1>,
              <h5
                className="header-description"
>
                Enter a location or check "use my location" to find out what's happening in your scene this week!
</h5>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Scene-It",
                "className": "App-title",
              },
              "ref": null,
              "rendered": "Scene-It",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Enter a location or check \\"use my location\\" to find out what's happening in your scene this week!",
                "className": "header-description",
              },
              "ref": null,
              "rendered": "Enter a location or check \\"use my location\\" to find out what's happening in your scene this week!",
              "type": "h5",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                id="use-current-location"
                onChange={[Function]}
                type="checkbox"
/>,
              <label
                htmlFor="use-current-location"
>
                Use my current location
</label>,
            ],
            "className": "user-select",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "use-current-location",
                "onChange": [Function],
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Use my current location",
                "htmlFor": "use-current-location",
              },
              "ref": null,
              "rendered": "Use my current location",
              "type": "label",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                id="locationName"
                onChange={[Function]}
                placeholder="location"
                value=""
/>,
              <input
                id="keywords"
                onChange={[Function]}
                placeholder="keywords (optional)"
/>,
              <button
                className="landing-submit"
                disabled={true}
>
                Get Events
</button>,
            ],
            "className": "query-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "locationName",
                "onChange": [Function],
                "placeholder": "location",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "keywords",
                "onChange": [Function],
                "placeholder": "keywords (optional)",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Get Events",
                "className": "landing-submit",
                "disabled": true,
              },
              "ref": null,
              "rendered": "Get Events",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="landing-form"
          onSubmit={[Function]}
>
          <header
                    className="App-header"
          >
                    <h1
                              className="App-title"
                    >
                              Scene-It
                    </h1>
                    <h5
                              className="header-description"
                    >
                              Enter a location or check "use my location" to find out what's happening in your scene this week!
                    </h5>
          </header>
          <div
                    className="user-select"
          >
                    <input
                              id="use-current-location"
                              onChange={[Function]}
                              type="checkbox"
                    />
                    <label
                              htmlFor="use-current-location"
                    >
                              Use my current location
                    </label>
          </div>
          <div
                    className="query-info"
          >
                    <input
                              id="locationName"
                              onChange={[Function]}
                              placeholder="location"
                              value=""
                    />
                    <input
                              id="keywords"
                              onChange={[Function]}
                              placeholder="keywords (optional)"
                    />
                    <button
                              className="landing-submit"
                              disabled={true}
                    >
                              Get Events
                    </button>
          </div>
</form>,
        "className": "landing-body",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <header
              className="App-header"
>
              <h1
                            className="App-title"
              >
                            Scene-It
              </h1>
              <h5
                            className="header-description"
              >
                            Enter a location or check "use my location" to find out what's happening in your scene this week!
              </h5>
</header>,
            <div
              className="user-select"
>
              <input
                            id="use-current-location"
                            onChange={[Function]}
                            type="checkbox"
              />
              <label
                            htmlFor="use-current-location"
              >
                            Use my current location
              </label>
</div>,
            <div
              className="query-info"
>
              <input
                            id="locationName"
                            onChange={[Function]}
                            placeholder="location"
                            value=""
              />
              <input
                            id="keywords"
                            onChange={[Function]}
                            placeholder="keywords (optional)"
              />
              <button
                            className="landing-submit"
                            disabled={true}
              >
                            Get Events
              </button>
</div>,
          ],
          "className": "landing-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="App-title"
>
                  Scene-It
</h1>,
                <h5
                  className="header-description"
>
                  Enter a location or check "use my location" to find out what's happening in your scene this week!
</h5>,
              ],
              "className": "App-header",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Scene-It",
                  "className": "App-title",
                },
                "ref": null,
                "rendered": "Scene-It",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Enter a location or check \\"use my location\\" to find out what's happening in your scene this week!",
                  "className": "header-description",
                },
                "ref": null,
                "rendered": "Enter a location or check \\"use my location\\" to find out what's happening in your scene this week!",
                "type": "h5",
              },
            ],
            "type": "header",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  id="use-current-location"
                  onChange={[Function]}
                  type="checkbox"
/>,
                <label
                  htmlFor="use-current-location"
>
                  Use my current location
</label>,
              ],
              "className": "user-select",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "use-current-location",
                  "onChange": [Function],
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Use my current location",
                  "htmlFor": "use-current-location",
                },
                "ref": null,
                "rendered": "Use my current location",
                "type": "label",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  id="locationName"
                  onChange={[Function]}
                  placeholder="location"
                  value=""
/>,
                <input
                  id="keywords"
                  onChange={[Function]}
                  placeholder="keywords (optional)"
/>,
                <button
                  className="landing-submit"
                  disabled={true}
>
                  Get Events
</button>,
              ],
              "className": "query-info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "locationName",
                  "onChange": [Function],
                  "placeholder": "location",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "keywords",
                  "onChange": [Function],
                  "placeholder": "keywords (optional)",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Get Events",
                  "className": "landing-submit",
                  "disabled": true,
                },
                "ref": null,
                "rendered": "Get Events",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
